github/with_appengine.go

```diff
@@ -0,0 +1,21 @@
+// Copyright 2017 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// This file provides glue for making github work on App Engine.
+
+// +build appengine
+
+package github
+
+import (
+	"context"
```

Comment:
I don't use App Engine, so I'm not familiar. If https://github.com/google/go-github/issues/526#issuecomment-284177413 is accurate, and App Engine uses/forces you to use Go 1.6, how does this work? Package `context` was added to Go standard library in Go 1.7, it wasn't available in Go 1.6.